src/psyclone/psyir/nodes/if_block.py

```diff
@@ -89,20 +91,31 @@ def _validate_child(position, child):
             position in (1, 2) and isinstance(child, Schedule))
 
     @property
-    def condition(self):
+    def condition(self) -> DataNode:
         ''' Return the PSyIR Node representing the conditional expression
         of this IfBlock.
 
-        :returns: IfBlock conditional expression.
-        :rtype: :py:class:`psyclone.psyir.nodes.Node`
-        :raises InternalError: If the IfBlock node does not have the correct \
+        :returns: The conditional expression of this IfBlock.
+
+        :raises InternalError: If the IfBlock node does not have the correct
             number of children.
+        :raises InternalError: If the conditional expression is known to be of
+            the wrong type.
+
         '''
         if len(self.children) < 2:
             raise InternalError(
                 f"IfBlock malformed or incomplete. It should have at least 2 "
                 f"children, but found {len(self.children)}.")
-        return self._children[0]
+
+        cond = self._children[0]
+        if isinstance(cond, Literal) and (cond.datatype.intrinsic
```

Comment:
This seems to be unrelated to this PR?? Not sure if I am missing something.
I am also not quite certain if that is the right place to test this, shouldn't that be when creating the node? After all, it might be that a user script just uses `if_node.children[0]` (yes, it's bad of course ;) ), but I would try to prevent this from happening in the first place.